src/sources/fonts.ts

```diff
@@ -102,16 +103,16 @@ export default function getFonts(): Promise<string[]> {
 
     // creates spans for the base fonts and adds them to baseFontsDiv
     const initializeBaseFontsSpans = () => {
-      return baseFonts.map(createSpan)
+      return mapWithBreaks(baseFonts, createSpan)
     }
 
     // creates spans for the fonts to detect and adds them to fontsDiv
-    const initializeFontsSpans = () => {
+    const initializeFontsSpans = async () => {
       // Stores {fontName : [spans for that font]}
       const spans: Record<string, HTMLSpanElement[]> = {}
 
       for (const font of fontList) {
-        spans[font] = baseFonts.map((baseFont) => createSpanWithFonts(font, baseFont))
+        spans[font] = await mapWithBreaks(baseFonts, (baseFont) => createSpanWithFonts(font, baseFont))
```

Comment:
There are only 3 `baseFonts`, so `mapWithBreaks` should be excessive here. Same on the line 106.

Reply:
It makes sense to use `mapWithBreaks` for the `for (const font of fontList)` loop instead.

Reply:
I'm concerned about many microtasks because they can harm the performance. What do you think if we just release the loop once before measuring the DOM elements (both in `fonts` and `font_preferences`)?